account_easy_reconcile/base_reconciliation.py

```diff
@@ -172,33 +173,30 @@ def _reconcile_lines(self, cr, uid, rec, lines, allow_partial=False,
                  the second is wether the reconciliation is full (True)
                  or partial (False)
         """
-        if context is None:
-            context = {}
-        ml_obj = self.pool.get('account.move.line')
-        writeoff = rec.write_off
+        self.ensure_one()
+        ml_obj = self.env['account.move.line']
+        writeoff = self.write_off
         line_ids = [l['id'] for l in lines]
         below_writeoff, sum_debit, sum_credit = self._below_writeoff_limit(
-            cr, uid, rec, lines, writeoff, context=context)
-        date = self._get_rec_date(
-            cr, uid, rec, lines, rec.date_base_on, context=context)
-        rec_ctx = dict(context, date_p=date)
+            lines, writeoff
+        )
+        date = self._get_rec_date(lines, self.date_base_on)
+        rec_ctx = dict(self.env.context or {}, date_p=date)
```

Comment:
`self.env.context` should always be a frozen dict. Thus I don't think you need the `or {}`